src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import duke.command.Command;
+import duke.task.TaskList;
+
+import java.io.IOException;
+import java.nio.file.NoSuchFileException;
+
+public abstract class Duke {
+    protected static final String DEFAULT_FILE_PATH = "tasks.txt";
+    private Parser parser;
+    protected Ui ui;
+
+    /**
+     * Constructs a new copy of the Duke application.
+     *
+     * @param filePath File path of the save file to store tasks.
+     */
+    protected Duke(String filePath, Ui ui) {
+        this.ui = ui;
+        ui.showWelcome();
+        Storage storage = new Storage(filePath);
+        TaskList tasks;
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException | IOException e) {
+            if (!(e instanceof NoSuchFileException)) {
```

Comment:
Can consider using && to minimise arrowheading 
e.g. if((!(e instanceof NoSuchFileException)) && e instanceof DukeException)